### PR TITLE
Fix #2339: mark RESTv1 API as deprecated (as well as in-coordinator APIs)

### LINKED ISSUE #2339: Mark REST v1 endpoints as Deprecated
Based on discussion #2242, document the REST /v1 endpoints as being deprecated and scheduled for removal in Stargate v3.

Update documentation
- https://github.com/stargate/stargate/blob/main/restapi/README-docsapi.md 

Code warnings
- add startup message to restapi module at WARN level

### PR DESCRIPTION
**Pull request by tatu-at-datastax**
**What this PR does**:

Improves READMEs for in-coordinator (Stargate V1) REST and Documents APIs and adds warning logs for deprecation wrt these APIs.

NOTE: this describes the situation once all removal work has been completed wrt Stagate V2.
This will mean removal of conditional registration as well as implementations of in-coordinator DocsV2 and RESTv2 APIs.

**Which issue(s) this PR fixes**:
Fixes #2339

**Checklist**
- [x] Changes manually tested
- [ ] Automated Tests added/updated
- [ ] Documentation added/updated
- [ ] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: restapi/src/main/java/io/stargate/web/impl/RestApiServer.java [hunk 1]
```diff
@@ -170,16 +172,18 @@ protected void configure() {
 
     if (!enableSgv1) {
```
Comment (tatu-at-datastax) [inline restapi/src/main/java/io/stargate/web/impl/RestApiServer.java:173]: NOTE: we will soon simply remove this conditional inclusion and drop registration of in-coordinator DocsV2, RESTv1. But README change can go in first; we are only keeping implementations to allow quick and safe rollback of the very first StagateV2 production deployment, if necessary.

Comment (jeffreyscarpenter) [inline restapi/src/main/java/io/stargate/web/impl/RestApiServer.java:173]: Do we document this behavior anywhere? Should be in the README?

Comment (tatu-at-datastax) [reply restapi/src/main/java/io/stargate/web/impl/RestApiServer.java:173]: Ok. Yes, I will update this in a bit. Even if it's transient, makes sense.

Comment (tatu-at-datastax) [reply restapi/src/main/java/io/stargate/web/impl/RestApiServer.java:173]: @jeffreyscarpenter Ok, added a note on opt-in property. I hope this now works better?

File: restapi/README.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
+# Stargate V2 in-Coordinator RESTv1 API
+
+## Differences from Stargate V1
+
+Stargate V1 used to provide three in-Coordination API implementations via this module: Documents API (v2), REST API v1 and REST API v2).
```
Comment (ivansenic) [inline restapi/README.md:5]: extra `)` at the end?

File: restapi/src/main/java/io/stargate/web/impl/RestApiServer.java [hunk 3]
```diff
@@ -170,16 +172,18 @@ protected void configure() {
...
               : String.format("set to '%s'", enableSgv1RestStr));
     } else {
-      logger.info(
-          "Registering StargateV1 Documents API and RESTv2 endpoints for StargateV2 (System property '{}' set to '{}')",
+      logger.warn(
+          "Registering in-Coordinator Documents API and RESTv2 API endpoints (System property '{}' set to '{}')",
           SYSPROP_ENABLE_SGV1_REST,
           enableSgv1RestStr);
+      logger.warn(
+          "NOTE: in-Coordinator Documents API and RESTv2 APIs are DEPRECATED in Stargate V2 and scheduled for removal in V3");
```
Comment (ivansenic) [inline restapi/src/main/java/io/stargate/web/impl/RestApiServer.java:186]: isn't this stated in the readme to be removed in one of the 2.x releases?

**Review by ivansenic — APPROVED** [review]

File: restapi/README.md (hunk 2 above)
Comment (tatu-at-datastax) [reply restapi/README.md:5]: Thanks, will remove.

File: restapi/src/main/java/io/stargate/web/impl/RestApiServer.java (hunk 3 above)
Comment (tatu-at-datastax) [reply restapi/src/main/java/io/stargate/web/impl/RestApiServer.java:186]: Thanks. I had an earlier version with this description which is wrong; yes, will be removed.

**Pull request merged**